src/NServiceBus.AcceptanceTests/Core/Sagas/When_using_ReplyToOriginator_with_headers.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Sagas
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_using_ReplyToOriginator_with_headers : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_send_headers()
+        {
+            var customeHeaderValue = Guid.NewGuid();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When(session => session.SendLocal(new InitiateRequestingSaga
+                {
+                    CustomHeaderValue = customeHeaderValue
+                })))
+                .Done(c => c.CustomHeaderOnReply != null)
+                .Run();
+
+            Assert.AreEqual(customeHeaderValue.ToString(), context.CustomHeaderOnReply, "Header values should be forwarded");
+        }
+
+        public class Context : ScenarioContext
+        {
+            public string CustomHeaderOnReply { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                        config.LimitMessageProcessingConcurrencyTo(1) //to avoid race conditions with the start and second message
+                );
+            }
+
+            public class RequestingSaga : Saga<RequestingSaga.RequestingSagaData>,
```

Comment:
nitpick but not sure about the naming here. What does this saga request? This is more something like a `ReplyingSaga` isn't it?

Reply:
Fixed